src/packagedcode/go_mod.py

```diff
@@ -250,15 +250,19 @@ def parse_gomod(location):
 # Regex expressions to parse go.sum file dependency
 # dep example: github.com/BurntSushi/toml v0.3.1 h1:WXkYY....
 get_dependency = re.compile(
-    r'(?P<namespace>[^\s]*)'
-    r'\/'
     r'(?P<name>[^\s]*)'
     r'\s*'
     r'(?P<version>[^\s]*)'
     r'\s*'
     r'h1:(?P<checksum>[^\s]*)\s*'
 ).match
 
+name_parser = re.compile(
+    r'(?P<namespace>[^\s]*)'
+    r'\/'
+    r'(?P<name>[^\s]*)'
+).match
+
```

Comment:
I think I get your approach: get a `namespace/name` first and if it contains a `/` further split it afterwards below, right?
If so, let me suggest a slightly simpler approach for the second part below which would mean you do not need that second regex above